numpy/__init__.pyi

```diff
@@ -358,6 +358,187 @@ class ndarray(Iterable, Sized, SupportsInt, SupportsFloat, SupportsComplex,
     def __getattr__(self, name) -> Any: ...
 
 
+_Scalar = TypeVar("_Scalar", bound="generic")
+
+class generic:
+    @property
```

Comment:
Could we define a private base class for all the shared methods/properties between `generic` and `ndarray`? Otherwise we'll have quite a bit to keep in sync.